ci-operator/step-registry/cucushift/chainupgrade/toimage/cucushift-chainupgrade-toimage-commands.sh

```diff
@@ -618,6 +618,25 @@ function admin_ack() {
 
 # Upgrade the cluster to target release
 function upgrade() {
+    set_channel $TARGET_VERSION
+    local retry=3 conditional_updates
+    while (( retry > 0 )); do
```

Comment:
The similar comments to https://github.com/openshift/release/pull/50446#discussion_r1547164336, if here we are using while loop, I guess https://github.com/openshift/release/pull/50446/files#diff-b1c231163eb97f0cd23233ca9867c3a719e46d494dc44546166ac0074128579dR46 also need a similar loop.

Reply:
+1, except for this one, lgtm

Reply:
for `cucushift-upgrade-check-not-recommended`, we definitely need to exit it as failure, not just copy-paste the same codes.

Reply:
Wait, let me try to understand your comment.

Reply:
I think you are right, this step is to check `not recommended`, so if the target version is not recommended we should exit 1

Reply:
Yeah, the purpose of the not-recommended step is for exit in advance, so I think it should be work just copying the same code. But when you said how to add loop, I was confused, I thought I missed something.

Now it looks good to me.